web/source/osk/banner.ts

```diff
@@ -127,13 +128,124 @@ namespace com.keyman.osk {
     }
   }
 
+  export class OSKSuggestionSpec {
+    id: string;
+    languageID: string;
+    text?: string;
+    width: string;
+    pad?: string;
+    widthpc?: number; // Added during OSK construction.
+    padpc?: number; // Added during OSK construction.
+
+    constructor(id: string, languageID: string, text?: string, width?: string, pad?: string) {
+      this.id = id;
+      this.languageID = languageID;
+      this.text = text;
+      this.width = width ? width : "50";
+      this.pad = pad;
+    }
+  }
+
+  export class OSKSuggestion {
+    spec: OSKSuggestionSpec;
```

Comment:
While it's not the most important detail, I'd rather drop the `OSK` prefix on both, since it's more linked to the `Banner` instead.

Reply:
I ended up using a `Banner` prefix to avoid collisions with just `Suggestion`.